### PR TITLE
Added mongock.enabled condition to configuration

### LINKED ISSUE #225: Mongock should not create changeLog and lock collection when enabled = false
I was testing the 4.1.8 version of Mongock in Spring Boot, and I discovered that the Mongock collections were created (though empty) even when `spring.mongock.enabled` is `false`.  This is actually just fine for my use case, but it did surprise me.  Is it intended that the `mongockChangeLog` & `mongockLock` collections be created even when the functionality is disabled?

```
> show collections
[...]
mongockChangeLog
mongockLock
[...]
> db.mongockChangeLog.count()
0
> db.mongockLock.count()
0
```

### application.yml
```
  mongock:
    enabled: false
    change-logs-scan-package: com.example.my.migrations
    legacy-migration:
      collection-name: dbchangelog
```

### Log output:
```
17:10:01.567 [main] INFO  io.changock.runner.core.ChangockBase - Changock is disabled. Exiting.
```

### LINKED ISSUE #256: Mongock should not create changeLog and lock collection when enabled = false
I was testing the 4.1.8 version of Mongock in Spring Boot, and I discovered that the Mongock collections were created (though empty) even when `spring.mongock.enabled` is `false`.  This is actually just fine for my use case, but it did surprise me.  Is it intended that the `mongockChangeLog` & `mongockLock` collections be created even when the functionality is disabled?

```
> show collections
[...]
mongockChangeLog
mongockLock
[...]
> db.mongockChangeLog.count()
0
> db.mongockLock.count()
0
```

### application.yml
```
  mongock:
    enabled: false
    change-logs-scan-package: com.example.my.migrations
    legacy-migration:
      collection-name: dbchangelog
```

### Log output:
```
17:10:01.567 [main] INFO  io.changock.runner.core.ChangockBase - Changock is disabled. Exiting.
```

### PR DESCRIPTION
**Pull request by santam85**
Updated prefixes to match configuration

Fixes flamingock/mongock#256

### CODE REVIEW

**Comment by dieppa** [comment]
Hello @santam85 ,
thanks for your contribution! It looks good to me, I will review it asap, but I am missing some test. If you cannot make it, please let me know and we add it. But tests must be in place before merging.

**Comment by santam85** [comment]
This might not be the test you had in mind, since I ended up testing Spring's `@Conditional` in there (pretty useless test if you ask me), but I couldn't think of a better lightweight method of checking this configuration's effectiveness.

File: mongock-spring-v5/src/main/java/com/github/cloudyrock/spring/v5/MongockTestConfiguration.java [hunk 1]
```diff
@@ -1,16 +0,0 @@
...
-import java.lang.annotation.Retention;
-import java.lang.annotation.RetentionPolicy;
-import java.lang.annotation.Target;
-
-@Documented
-@Target(ElementType.TYPE)
-@Retention(RetentionPolicy.RUNTIME)
-@Import({MongockConfiguration.class, MongockTestContext.class})
-public @interface MongockTestConfiguration {
-}
```
Comment (dieppa) [inline mongock-spring-v5/src/main/java/com/github/cloudyrock/spring/v5/MongockTestConfiguration.java:16]: Why is this removed?

File: mongock-spring-v5/src/main/java/com/github/cloudyrock/spring/v5/MongockTestContext.java [hunk 2]
```diff
@@ -1,41 +0,0 @@
...
-    private final MongockConnectionDriver driver;
-    private final MongoTemplate mongoTemplate;
-
-    private MongockTestDriverInitializingBean(MongockConnectionDriver driver,
-                                              MongoTemplate mongoTemplate) {
-      this.driver = driver;
-      this.mongoTemplate = mongoTemplate;
-    }
-
-    @Override
```
Comment (dieppa) [inline mongock-spring-v5/src/main/java/com/github/cloudyrock/spring/v5/MongockTestContext.java:34]: Why is this removed?

File: mongock-spring-v5/src/main/java/com/github/cloudyrock/spring/v5/MongockSpringDataV2CoreContext.java [hunk 3]
```diff
@@ -19,7 +21,7 @@ public SpringDataMongo2Driver mongockConnectionDriver(MongoTemplate mongoTemplat
                                                         Optional<MongoTransactionManager> txManagerOpt) {
     try {
       SpringDataMongo2Driver driver = SpringDataMongo2Driver.withLockSetting(mongoTemplate, mongockConfiguration.getLockAcquiredForMinutes(), mongockConfiguration.getMaxWaitingForLockMinutes(), mongockConfiguration.getMaxTries());
-      if(mongockConfiguration.isTransactionEnabled() && txManagerOpt.isPresent()) {
+      if (mongockConfiguration.isTransactionEnabled() && txManagerOpt.isPresent()) {
```
Comment (dieppa) [inline mongock-spring-v5/src/main/java/com/github/cloudyrock/spring/v5/MongockSpringDataV2CoreContext.java:24]: thanks

File: mongock-spring-v5/src/test/java/com/github/cloudyrock/spring/v5/MongockContextTest.java [hunk 4]
```diff
@@ -0,0 +1,51 @@
+package com.github.cloudyrock.spring.v5;
```
Comment (dieppa) [inline mongock-spring-v5/src/test/java/com/github/cloudyrock/spring/v5/MongockContextTest.java:1]: As you mentioned, it doesn't prove the bean is not injected due to enabled=false.
We'd need to prove the base condition, meaning that the beans are injected when enabled is true or missing.

**Review by dieppa — CHANGES_REQUESTED** [review]
Nice work, thanks. I left some comments.
My main concerns are two: (1) why some files are removed and (2)the real value of the tests proving(and ensuring) that enabled=false will prevent beans from being injected and enabled=true(or missing) will inject them.

Thanks!👍🏽

File: mongock-spring-v5/src/test/java/com/github/cloudyrock/spring/v5/MongockContextTest.java (hunk 4 above)
Comment (santam85) [reply mongock-spring-v5/src/test/java/com/github/cloudyrock/spring/v5/MongockContextTest.java:1]: What you would be testing in a way would then be the effectiveness of the `@Conditional` annotation, which I am sure the Spring team has done extensively.

Due to the hard dependencies of the beans to MongoTemplate, I would need to stand up a full App Context (and likely test containers) to have a working e2e test, for instance with a `@SpringBootTest` annotation to keep it concise. Is this what you would like?

File: mongock-spring-v5/src/main/java/com/github/cloudyrock/spring/v5/MongockTestContext.java (hunk 2 above)
Comment (santam85) [reply mongock-spring-v5/src/main/java/com/github/cloudyrock/spring/v5/MongockTestContext.java:34]: It was unused. If we want to keep it, I believe it should be part of the test classpath.

File: mongock-spring-v5/src/main/java/com/github/cloudyrock/spring/v5/MongockTestConfiguration.java (hunk 1 above)
Comment (santam85) [reply mongock-spring-v5/src/main/java/com/github/cloudyrock/spring/v5/MongockTestConfiguration.java:16]: It was unused. If we want to keep it, I believe it should be part of the test classpath.

**Comment by santam85** [comment]
Would you consider using `@ConditionalOnBean(value = MongoTemplate.class)` instead of the @EnableMongock annotation and together with the enable/disable property in order to trigger Mongock? It would simplify testing scenarios and configuration.

**Comment by dieppa** [comment]
> Would you consider using `@ConditionalOnBean(value = MongoTemplate.class)` instead of the @EnableMongock annotation and together with the enable/disable property in order to trigger Mongock? It would simplify testing scenarios and configuration.

I see your point, but first I don't see the configuration being complex to be honest, but probably I am not the best person to judge this. However the most important fact to keep`@EnableMongock` is that users won't alway us the properties approach, sometimes they will want to use the traditional builder approach. Your suggestion means the user would need to add `enabled=false` to the properties file if they want to use the builder, which I see unnecessary and confusing.

Do you see my point?

File: mongock-spring-v5/src/main/java/com/github/cloudyrock/spring/v5/MongockTestContext.java (hunk 2 above)
Comment (dieppa) [reply mongock-spring-v5/src/main/java/com/github/cloudyrock/spring/v5/MongockTestContext.java:34]: It's requires in some test scenario, when they want Mongock to be enabled but still want to call the changeSet directly. It's an edge case for a user, but still relevant. I rather keep it ;)

File: mongock-spring-v5/src/main/java/com/github/cloudyrock/spring/v5/MongockTestConfiguration.java (hunk 1 above)
Comment (dieppa) [reply mongock-spring-v5/src/main/java/com/github/cloudyrock/spring/v5/MongockTestConfiguration.java:16]: It's requires in some test scenario, when they want Mongock to be enabled but still want to call the changeSet directly. It's an edge case for a user, but still relevant. I rather keep it ;)

File: mongock-spring-v5/src/test/java/com/github/cloudyrock/spring/v5/MongockContextTest.java (hunk 4 above)
Comment (dieppa) [reply mongock-spring-v5/src/test/java/com/github/cloudyrock/spring/v5/MongockContextTest.java:1]: Maybe I am missing something here, butifarraI don't see why you need TestContainers. As we are not performing any database action or check, we can mock it.

The only issue I think you could have is related to both drivers, mongodb-springdata-v3-driver and mongodb-springdata-v2-driver, as you would probably need to import one of them.

If it's too painful, don't worry, I do it. we have a separate project, one per driver, for integration tests.
But let me know

**Comment by santam85** [comment]
> > Would you consider using `@ConditionalOnBean(value = MongoTemplate.class)` instead of the @EnableMongock annotation and together with the enable/disable property in order to trigger Mongock? It would simplify testing scenarios and configuration.
> 
> I see your point, but first I don't see the configuration being complex to be honest, but probably I am not the best person to judge this. However the most important fact to keep`@EnableMongock` is that users won't alway us the properties approach, sometimes they will want to use the traditional builder approach. Your suggestion means the user would need to add `enabled=false` to the properties file if they want to use the builder, which I see unnecessary and confusing.
> 
> Do you see my point?

At the moment, `enabled=false` is needed anyway for unit tests in applications initializing an `ApplicationContext` slice that does not include `MongoTemplate`, so I would argue both are needed.

**Comment by dieppa** [comment]
> > > Would you consider using `@ConditionalOnBean(value = MongoTemplate.class)` instead of the @EnableMongock annotation and together with the enable/disable property in order to trigger Mongock? It would simplify testing scenarios and configuration.
> > 
> > 
> > I see your point, but first I don't see the configuration being complex to be honest, but probably I am not the best person to judge this. However the most important fact to keep`@EnableMongock` is that users won't alway us the properties approach, sometimes they will want to use the traditional builder approach. Your suggestion means the user would need to add `enabled=false` to the properties file if they want to use the builder, which I see unnecessary and confusing.
> > Do you see my point?
> 
> At the moment, `enabled=false` is needed anyway for unit tests in applications initializing an `ApplicationContext` slice that does not include `MongoTemplate`, so I would argue both are needed.

well, having @Enabled in the Spring boot application in production code and mark it as disabled for some tests you don't want  Mongock to run, doesn't seem that bad.

**Comment by santam85** [comment]
Could not make it work, will wait for someone else to figure it out...

**Comment by dieppa** [comment]
sorry, what exactly you could not make it work?

**Comment by santam85** [comment]
Could not figure out how to exclude one of the 2 drivers from a test application, they share similar package names and classes. Also, there seems to be a mismatch between the drivers versions in the specified versions of spring boot and spring data.

I will go ahead and create a starter for myself that embraces a more opinionated spring-boot-starter approach, the way the current mongock-spring-v5 module works does not fully fit my need. Will perhaps make a PR with that.
